File: backend/test_supabase_connection.py
#!/usr/bin/env python3
"""
Test Supabase connection and debug issues
"""

import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

print("=== Supabase Connection Test ===")
print(f"SUPABASE_URL: {os.getenv('SUPABASE_URL')}")
print(f"SUPABASE_SERVICE_ROLE_KEY: {os.getenv('SUPABASE_SERVICE_ROLE_KEY')[:20] + '...' if os.getenv('SUPABASE_SERVICE_ROLE_KEY') else 'None'}")

try:
    from supabase import create_client, Client
    print("✅ Supabase package imported successfully")
    
    supabase_url = os.getenv("SUPABASE_URL")
    supabase_key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")
    
    if not supabase_url or not supabase_key:
        print("❌ Missing Supabase credentials")
    else:
        print("✅ Supabase credentials found")
        
        try:
            client = create_client(supabase_url, supabase_key)
            print("✅ Supabase client created successfully")
            
            # Test a simple query
            result = client.table("geographic_bias_analyses").select("id", count="exact").execute()
            print(f"✅ Database query successful: {result.count} records found")
            
        except Exception as e:
            print(f"❌ Failed to create Supabase client: {e}")
            
except ImportError as e:
    print(f"❌ Failed to import Supabase: {e}")
except Exception as e:
    print(f"❌ Unexpected error: {e}")

File: tools/scripts/test-supabase-connection.js
#!/usr/bin/env node
/**
 * Test Supabase connection using the correct environment file path
 */

const { createClient } = require('@supabase/supabase-js');
const dotenv = require('dotenv');
const path = require('path');

// Load environment variables from the correct path
dotenv.config({ path: path.resolve(__dirname, '../../.env.local') });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

console.log('=== Supabase Connection Test ===');
console.log(`SUPABASE_URL: ${supabaseUrl}`);
console.log(`SUPABASE_ANON_KEY: ${supabaseKey ? supabaseKey.substring(0, 20) + '...' : 'None'}`);

if (!supabaseUrl || !supabaseKey) {
  console.error('❌ Error: Missing Supabase URL or Anon Key in environment variables');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function testSupabaseConnection() {
  try {
    console.log('\n🔍 Testing Supabase connection...');
    
    // Test basic connection by trying to list tables
    const { data: tables, error } = await supabase
      .from('information_schema.tables')
      .select('table_name')
      .eq('table_schema', 'public')
      .limit(5);
    
    if (error) {
      console.error('❌ Error connecting to Supabase:');
      console.error(error);
      return;
    }
    
    console.log('✅ Successfully connected to Supabase!');
    
    if (tables && tables.length > 0) {
      console.log('\n📋 Found tables in public schema:');
      tables.forEach(table => {
        console.log(`  - ${table.table_name}`);
      });
    } else {
      console.log('\nℹ️  No tables found in public schema.');
    }
    
    // Test storage functionality
    console.log('\n--- Testing Storage ---');
    try {
      const { data: buckets, error: storageError } = await supabase.storage.listBuckets();
      
      if (storageError) {
        console.error('❌ Error accessing storage:', storageError.message);
      } else {
        console.log(`✅ Storage access successful. Found ${buckets.length} buckets:`);
        buckets.forEach(bucket => {
          console.log(`  - ${bucket.name} (public: ${bucket.public})`);
        });
        
        // Check if ai-models bucket exists
        const aiModelsBucket = buckets.find(b => b.name === 'ai-models');
        if (aiModelsBucket) {
          console.log('✅ ai-models bucket exists');
        } else {
          console.log('❌ ai-models bucket does not exist');
        }
      }
    } catch (storageErr) {
      console.error('❌ Storage test failed:', storageErr.message);
    }
    
  } catch (err) {
    console.error('❌ Unexpected error:');
    console.error(err);
  }
}

testSupabaseConnection();
